### ISSUE TITLE
Reports > Sales > "How to expand all"

### ISSUE THREAD

**Issue by bublers** [jekkos/opensourcepos#1009]
I am very new to GITHUB but a fan of OpensourcePOS
Previously in 2.x we can click "expand all" in Sales report.

1) Is there any how I can export detailed(expanded) to excel format
or
2) How do we bring back the "expand all" button in report page ?

Thank you !

**Comment by daN4cat** [comment]
1. Have click the square buttons on the top right of the table? One is to report the data

**Comment by bublers** [comment]
Thanks daN4cat for the help ...
Here some attachment 

I dint see any related button to "expand all" receipt details like previously does ..
Attached screenshot showing to the "expand all button Im referring to

![yaka2](https://cloud.githubusercontent.com/assets/23728400/20610149/a789da60-b2d0-11e6-81b7-be4aba011b16.PNG)

![yaka](https://cloud.githubusercontent.com/assets/23728400/20610051/64912084-b2cf-11e6-93e8-b2cb5ed48204.PNG)

**Comment by daN4cat** [comment]
Yes I know, that was point 2), there is not such function at the moment.

**Comment by bublers** [comment]
Ah ... thank you daN4cat !

**Comment by daN4cat** [comment]
@jekkos not sure if there is any support in bstables for the expand all feature?

**Comment by jekkos** [comment]
[There is .]( https://github.com/wenzhixin/bootstrap-table/pull/1435)

**Comment by daN4cat** [comment]
adding on top of the table sorting?

**Comment by jekkos** [comment]
Not sure I'll have to look into that.

**Comment by amrshah** [comment]
I am having this output on my sale report :-S
![sales report error](https://user-images.githubusercontent.com/1771447/29467701-db8cacdc-845a-11e7-94d0-dbedf0a5d7e3.png)

**Comment by amrshah** [comment]
Looks like there's some bug because of which the header column names are not being printed correctly!

Could that relate to the language file issue?

**Comment by amrshah** [comment]
![sales report problem](https://user-images.githubusercontent.com/1771447/29468646-8bbfd248-845e-11e7-9ef0-fcc4e574a27a.png)

**Comment by amrshah** [comment]
Same things is happening on all my reports Customer report,  employees report etc

**Comment by odiea** [comment]
Yes . If you are using 3.0.2 it does not support php 7 or mysql 5.7.

**Comment by amrshah** [comment]
Ahan....okay. Thank you @odiea

**Comment by daN4cat** [comment]
Why are we posting unrelated stuff randomly? I'll delete this later.

**Comment by odiea** [comment]
Deleted unrelated issues.

**Comment by mambiux** [comment]
Is there any way I can export detailed(expanded) to excel format, including all items for each sale?, been thinkering for days without success so far. Please help, and thank you, LOVE OSPOS

**Comment by SteveIreland** [comment]
I think this thread is cursed.

**Comment by streetlife** [comment]
The export functions uses the Bootstrap table as opposed to exporting from the database.  Hence it doesn't export the sub-tables.
